### DISCUSSION TITLE
MAP-MRI optimization & validation errors

### DISCUSSION THREAD

**Discussion by l-espana** [dipy/dipy#3575] [Reconstruction] [answered]
I haven't found mention of this issue anywhere so apologies if I'm missing something. We're using the basic reconst mapmri example script (with a few changes) on some of our data, but some subjects have the following errors/warnings:

/usr/local/lib/python3.6/dist-packages/dipy/reconst/mapmri.py:396: RuntimeWarning: invalid value encountered in true_divide
  data_norm = np.asarray(data / data[self.gtab.b0s_mask].mean())
/usr/local/lib/python3.6/dist-packages/dipy/reconst/mapmri.py:412: UserWarning: Optimization did not find a solution
  warn('Optimization did not find a solution')
/usr/local/lib/python3.6/dist-packages/dipy/reconst/mapmri.py:396: RuntimeWarning: divide by zero encountered in true_divide
  data_norm = np.asarray(data / data[self.gtab.b0s_mask].mean())
map.py:133: RuntimeWarning: invalid value encountered in power
  rtopCbRt = rtop**(1/3.0)
map.py:138: RuntimeWarning: invalid value encountered in sqrt
  rtapSqRt = np.sqrt(rtap)
Preprocessing: 20018
Running model
>> data.shape (256, 256, 30, 96)
>> reconstruction begin time  0.1 min
>> data_small.shape (256, 256, 30, 96)
>> mask_roi.shape (256, 256, 30)
**DENOISE PROCESS
>> No denoising applied
**MAP MODEL
>> MAP model fit, part1 begin time  0.1 min
>> Applying both laplacian regularization and positivity constraint
ERROR in validate_data: Lower bound at index 1 is greater than upper bound: inf > 1.0000e+20
ERROR in osqp_setup: Data validation returned failure
>> MAP model fit, part1 completed  3941.4 min
**CALCULATE PROPERTIES
>> Return to Origin Probability
>> Return to Axis Probability
>> Return to Plane Probability
>> Mean Square Displacement
~                           

There still appears to be output, but the validation errors seem concerning.

Running Ubuntu 16.04, python 3.6, dipy 0.16.0. I'm not sure what else may be helpful, but any thoughts on this would be much appreciated.

**Comment by arokem** [comment, marked as answer]
Without knowing more, I am going to guess that these warnings are not of major concern. My intuition is that they are coming from voxels in your data that contain physically implausible values, due to noise, or measurements in parts of the measurement that do not exhibit brain-like properties. This often happens in the skull or even outside the head, where all you are measuring is noise.

**Comment by l-espana** [comment]
Thanks for your response. Yes, the initial warnings would appear to be voxels outside the brain (despite using a brain mask as the mask_roi if I remember correctly), but could this also be the cause of the validation errors during the laplacian regularization and positivity constraint process? That or some other noise? Or is there possibly some other issue?
I did a quick visual inspection of subjects with and without these errors and didn't notice any differences, but I'm not sure if there's a more robust way to check.
